### ISSUE TITLE
reqd_work_group_size attribute is not reversed correctly

### ISSUE THREAD

**Issue by Pennycook** [intel/llvm#4224]
**Describe the bug**
Using `sycl::reqd_work_group_size` to specify a 1- or 2-dimensional work-group size compiles correctly, but fails at runtime.  Something somewhere is not correctly accounting for the difference in dimension ordering between SYCL and OpenCL.

**To Reproduce**
```c++
#include <CL/sycl.hpp>

int main()
{
  sycl::queue q;
  q.parallel_for(sycl::nd_range<1>{32, 16},
                 [=](sycl::nd_item<1> it) [[sycl::reqd_work_group_size(16)]] {});
  q.parallel_for(sycl::nd_range<2>{sycl::range<2>{32, 32}, sycl::range<2>{16, 16}},
                 [=](sycl::nd_item<2> it) [[sycl::reqd_work_group_size(16, 16)]] {});
}
```

The first kernel fails to launch, with the following error:
```
terminate called after throwing an instance of 'cl::sycl::nd_range_error'
  what():  The specified local size {16, 1, 1} doesn't match the required work-group size specified in the program source {1, 1, 16} -54 (CL_INVALID_WORK_GROUP_SIZE)
Aborted (core dumped)
```

If you comment out the first kernel, the second kernel fails to launch with a similar error:
```
terminate called after throwing an instance of 'cl::sycl::nd_range_error'
  what():  The specified local size {16, 16, 1} doesn't match the required work-group size specified in the program source {1, 16, 16} -54 (CL_INVALID_WORK_GROUP_SIZE)
Aborted (core dumped)
```

Note that in both cases the sizes are captured correctly, but appear in the wrong order (e.g. `{16, 16, 1}` vs `{1, 16, 16}`).

**Environment (please complete the following information):**

- OS: Linux
- Target device and vendor: Tested on Intel CPU and Intel GPU
- DPC++ version: Nightly from 20210729

**Additional context**
I couldn't find any tests for this functionality here or in intel/llvm-test-suite.  I found some Sema tests, but nothing that runs a kernel to check that the requested work-group size was respected by the backend.

**Comment by AaronBallman** [comment]
FWIW, I think this relates to https://github.com/intel/llvm/issues/3175 and https://github.com/intel/llvm/issues/3361.

**Comment by zjin-lcf** [comment]
I have a question. Does sycl::nd_range<1>{32, 16}  or sycl::nd_range<2>{sycl::range<2>{32, 32}, sycl::range<2>{16, 16}  specify the required work-group size(s) in the first place ?  Thanks.

**Comment by smanna12** [comment]
@Pennycook, i will take a look at this bug.

**Comment by AlexeySachkov** [comment]
I think that this was fixed by @steffenlarsen in #7450. @steffenlarsen - could you please confirm and close this issue?

**Comment by steffenlarsen** [comment]
https://github.com/intel/llvm/pull/7450 has indeed addressed this issue. Closing it.
